### PR TITLE
Store compatible Contao version range

### PR DESCRIPTION
**Pull request by ausi**
As discussed in yesterdays call, this adds two new fields:

```json
"contaoConstraint": ">= 4.4.4.0-dev < 5.0.0.0-dev || >= 5.2.0.0-dev < 6.0.0.0-dev",
"contaoVersions": ["4.4 +", "5.2 +"],
```

### CODE REVIEW

**Comment by aschempp** [comment]
Looks good to me. Did you manually test it? What would a complex condition look like? Also, I would prefer `4.13+` and not `4.13 +`. Or maybe

**Comment by ausi** [comment]
> What would a complex condition look like?

Like this: `>= 4.0.0.0-dev < 5.0.0.0-dev || == 5.0.0.0-RC1 || == 5.0.0.0-RC2 || == 5.0.0.0-RC3 || == 5.0.0.0-RC4 || == 5.0.0.0 || == 5.0.1.0 || == 5.0.2.0 || == 5.0.3.0 || == 5.0.4.0 || == 5.0.5.0 || == 5.0.6.0 || == 5.0.7.0 || == 5.0.8.0 || == 5.0.9.0 || == 5.0.10.0 || == 5.1.0.0-RC1 || == 5.1.0.0-RC2 || == 5.1.0.0-RC3 || == 5.1.0.0 || == 5.1.1.0 || == 5.1.2.0 || == 5.1.3.0 || == 5.1.4.0 || == 5.1.5.0 || == 5.1.6.0 || == 5.1.7.0 || == 5.1.8.0 || == 5.1.9.0 || == 5.1.10.0 || == 5.1.11.0 || == 5.2.0.0-RC1 || == 5.2.0.0-RC2 || == 5.2.0.0-RC3 || == 5.2.0.0-RC4 || == 5.2.0.0-RC5 || == 5.2.0.0-RC6 || == 5.2.0.0 || == 5.2.1.0 || == 5.2.2.0 || == 5.2.3.0 || == 5.2.4.0 || == 5.2.5.0 || == 5.2.6.0 || == 5.2.7.0 || == 5.2.8.0 || == 5.2.9.0 || == 5.2.10.0 || == 5.3.0.0-RC1 || == 5.3.0.0-RC2 || == 5.3.0.0-RC3 || == 5.3.0.0-RC4 || == 5.3.0.0 || == 5.3.1.0 || == 5.3.2.0 || == 5.3.3.0 || == 5.3.4.0 || == 5.3.5.0 || == 5.3.6.0 || == 5.3.7.0 || == 5.3.8.0 || == 5.3.9.0 || == 5.3.10.0 || == 5.3.11.0 || == 5.4.0.0-RC1 || == dev-release/4.3.0`
but this is just long, not really more complex, it is always a OR-list of ranges or exact versions.

**Comment by ausi** [comment]
And only very vew packages do not support an open range. One example is `metamodels/core`:
```json
"contaoConstraint": ">= 4.9.0.0-dev < 4.13.0.0-dev",
"contaoVersions": ["4.9 - 4.12"],
```

File: indexer/src/Package/Factory.php [hunk 1]
```diff
@@ -169,4 +179,77 @@ static function (?string $prev, string $curr) use ($versions) {
...
+                    $version = "$major.$lastMinor";
+                    $matches[] = $version === $prefix ? $version : "$prefix - $version";
+                    $prefix = '';
+                }
+
+                $lastMinor = $minor;
+            }
+
+            if ($prefix) {
+                $matches[] = "$prefix +";
```
Comment (ausi) [inline indexer/src/Package/Factory.php:249]:
```suggestion
                $matches[] = "$prefix+";
```

**Comment by ausi** [comment]
> Or maybe

Maybe what? ☺️

**Review by aschempp — APPROVED** [review]

**Comment by aschempp** [comment]
No idea about the "Or maybe" 😂 
Lets merge this and see what data we get in Algolia.

**Pull request merged**